watchtower-core/src/main/java/com/tracelink/appsec/watchtower/core/scan/image/service/ImageScanResultService.java

```diff
@@ -0,0 +1,191 @@
+import com.tracelink.appsec.watchtower.core.scan.image.entity.ImageScanEntity;
+import com.tracelink.appsec.watchtower.core.scan.image.entity.ImageViolationEntity;
+import com.tracelink.appsec.watchtower.core.scan.image.report.ImageScanError;
+import com.tracelink.appsec.watchtower.core.scan.image.repository.ImageContainerRepository;
+import com.tracelink.appsec.watchtower.core.scan.image.repository.ImageScanRepository;
+import com.tracelink.appsec.watchtower.core.scan.image.repository.ImageViolationRepository;
+import com.tracelink.appsec.watchtower.core.scan.image.result.ImageResultFilter;
+import com.tracelink.appsec.watchtower.core.scan.image.result.ImageScanResult;
+import com.tracelink.appsec.watchtower.core.scan.image.result.ImageScanResultViolation;
+import com.tracelink.appsec.watchtower.core.scan.repository.RepositoryRepository;
+
+/**
+ * Handles logic around storing a retrieving scan results
+ *
+ * @author csmith
+ */
+@Service
+public class ImageScanResultService
+		extends AbstractScanResultService<ImageScanEntity, ImageViolationEntity> {
+
+	private static final Logger LOG = LoggerFactory.getLogger(ImageScanResultService.class);
+
+	private final ImageContainerRepository containerRepo;
+	private final RepositoryRepository imageRepo;
+	private final ImageScanRepository scanRepo;
+	private final ImageViolationRepository vioRepo;
+
+	public ImageScanResultService(@Autowired ImageContainerRepository containerRepo,
+			@Autowired RepositoryRepository imageRepo, @Autowired ImageScanRepository scanRepo,
+			@Autowired ImageViolationRepository vioRepo) {
+		super(scanRepo, vioRepo);
+		this.containerRepo = containerRepo;
+		this.imageRepo = imageRepo;
+		this.scanRepo = scanRepo;
+		this.vioRepo = vioRepo;
+	}
+
+	/**
+	 * Counts the number of repositories that have been scanned
+	 *
+	 * @return number of repositories scanned
+	 */
+	public long countImages() {
+		return imageRepo.count();
+	}
+
+	/**
+	 * Counts the number of pull requests that have been scanned
+	 *
+	 * @return number of pull requests scanned
+	 */
+	public long countContainers() {
+		return this.containerRepo.count();
+	}
+
+	/**
+	 * Store an Image Report to the database
+	 * 
+	 * @param scan       the scan to save
+	 * @param startTime  the starting time of the scan
+	 * @param violations the violations found in this scan
+	 * @param errors     the errors found in this scan
+	 */
+	public void saveImageReport(ImageScan scan, long startTime,
+			List<ImageViolationEntity> violations,
+			List<ImageScanError> errors) {
+		if (violations == null) {
+			return;
+		}
+		long now = System.currentTimeMillis();
+		ImageScanContainerEntity imageEntity =
+				containerRepo.findOneByApiLabelAndRepositoryNameAndTagName(scan.getApiLabel(),
+						scan.getRepository(), scan.getTag());
+		if (imageEntity == null) {
+			imageEntity = new ImageScanContainerEntity(scan);
+			imageEntity = containerRepo.saveAndFlush(imageEntity);
+		}
+		// TODO should we set a resolved flag?
```

Comment:
Is this still an open question?

Reply:
hmmm... I added resolved to PRs because we could recover from downtime with it. I don't think we realistically can with this one... so probably not. I'll remove the comment